netwaystev2/src/filter/filter.rs

```diff
@@ -451,17 +465,87 @@ impl Filter {
                         server = other_end_server;
                     }
                 }
-                //XXX send NewGenStateDiff/NewGameUpdates/NewChats as needed (if new)
                 if let Some(ref mut room) = server.room {
                     if let Some(ref mut game) = room.game {
                         if let Some(gen_state_diff) = game.process_genstate_diff_part(universe_update)? {
-                            filter_notice_tx.send(FilterNotice::NewGenStateDiff{
-                                diff: gen_state_diff,
-                                endpoint: endpoint,
-                            }).await?;
+                            filter_notice_tx
+                                .send(FilterNotice::NewGenStateDiff {
+                                    diff: gen_state_diff,
+                                    endpoint,
+                                })
+                                .await?;
+                        }
+                    }
+                }
+
+                let mut start_idx = None;
+                match (server.game_update_seq, game_update_seq) {
```

Comment:
Can you leave a comment explaining what the purpose of this block is? This seems like one of those things that makes sense now but will be easy to forget after several months.